addons/calendar/models/calendar_event.py

```diff
@@ -769,10 +769,12 @@ def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=Fals
         grouped_fields = {group_field.split(':')[0] for group_field in groupby + fields_aggregates}
         private_fields = grouped_fields - self._get_public_fields()
         if not self.env.su and private_fields:
+            # Sub query user settings from calendars that are not private ('public' and 'confidential').
+            public_calendars_settings = self.env['res.users.settings'].sudo()._search([('calendar_default_privacy', '!=', 'private')])
             # display public, confidential events and events with default privacy when owner's default privacy is not private
             domain = AND([domain, [
                 '|', '|', '|', ('privacy', '=', 'public'), ('privacy', '=', 'confidential'), ('user_id', '=', self.env.user.id),
-                '&', ('privacy', '=', False), ('user_id.calendar_default_privacy', '!=', 'private')
+                '&', ('privacy', '=', False), ('user_id.res_users_settings_id', 'in', public_calendars_settings)
```

Comment:
I am afraid of the performances here if we have a lot of users and a lot of events :thinking: Did you do some tests before and after your PR to have an idea of the impact ? For example in a performance test or using the profiler.